appmonitoring/ts/src/CertificateGenerator.ts

```diff
@@ -318,58 +367,50 @@ export class CertificateManager {
             if (shouldRestartReplicaset) {
                 logger.info('Restarting ReplicaSet so the pods pick up new certificates...', operationId, this.requestMetadata);
-                await CertificateManager.RestartWebhookReplicaset(operationId, kc, clusterArmId, clusterArmRegion);
+                await CertificateManager.RestartWebhookDeployment(operationId, kc, clusterArmId, clusterArmRegion);
             }
         }
         else {
             logger.info('Nothing to do. All is good. Ending this run...', operationId, this.requestMetadata);
         }
     }
 
-    private static IsValidCertificate(operationId: string, mwhcCaBundle: string, webhookCertData: WebhookCertData, clusterArmId: string, clusterArmRegion: string): boolean {
-        try {
-            forge.pki.certificateFromPem(mwhcCaBundle);
-            forge.pki.certificateFromPem(webhookCertData.caCert);
-            forge.pki.certificateFromPem(webhookCertData.tlsCert);
-            forge.pki.privateKeyFromPem(webhookCertData.tlsKey);
-            return true;
-        } catch (error) {
-            logger.error('Error occured while trying to validate certificates!', operationId, this.requestMetadata);
-            logger.error(JSON.stringify(error), operationId, this.requestMetadata);
-            logger.SendEvent("CertificateValidationFailed", operationId, null, clusterArmId, clusterArmRegion, true, error);
-            return false;
-        }
-    }
-
-    private static async RestartWebhookReplicaset(operationId: string, kc: k8s.KubeConfig, clusterArmId: string, clusterArmRegion: string) {
-        kc.loadFromDefault();
-
-        const k8sApi = kc.makeApiClient(k8s.AppsV1Api);
-        const selector = "app-monitoring-webhook"
+    private static async RestartWebhookDeployment(operationId: string, kc: k8s.KubeConfig, clusterArmId: string, clusterArmRegion: string) {
         let name = null;
-
+    
         try {
-            const replicaSets: k8s.V1ReplicaSetList = (await k8sApi.listNamespacedReplicaSet(NamespaceName)).body;
+            const k8sApi = kc.makeApiClient(k8s.AppsV1Api);
+            const selector = "app-monitoring-webhook"
+            const deployments: k8s.V1DeploymentList = (await k8sApi.listNamespacedDeployment(NamespaceName)).body;
 
-            const matchingReplicaSets: k8s.V1ReplicaSet[] = replicaSets.items.filter(rs => 
-                rs.spec.selector && rs.spec.selector.matchLabels && selector.localeCompare(rs.spec.selector.matchLabels.app) === 0
+            if (!deployments)
+            {
+                throw new Error(`No Deployments found in ${NamespaceName} namespace!`);
+            }
+    
+            const matchingDeployments: k8s.V1Deployment[] = deployments.items.filter(deployment => 
+                deployment.spec.selector && deployment.spec.selector.matchLabels && selector.localeCompare(deployment.spec.selector.matchLabels.app) === 0
```

Comment:
We do know the webhook deployment is the only deployment with the same app label, correct?

Reply:
Yes